src/qutip_tensorflow/core/data/expect.py

```diff
@@ -0,0 +1,75 @@
+import qutip
+from math import sqrt
+from .tftensor import TfTensor
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import tensorflow as tf
+
+
+__all__ = ["expect_tftensor", "expect_super_tftensor"]
+
+
+def _check_shape_expect(op, state):
+    if (
+        op.shape[0] != op.shape[1]  # Op square matrix.
+        or not (
+            state.shape[0] == state.shape[1]  # state is square matrix
+            or state.shape[1] == 1  # state is ket
+        )
+        or op.shape[1] != state.shape[0]  # Not possible op@state
+    ):
+        raise ValueError(
+            f"incompatible matrix shapes for op {op.shape} and state {state.shape}"
+        )
+
+
+def _check_shape_super(op, state):
+    if state.shape[1] != 1:
+        raise ValueError(
+            "expected a column-stacked matrix but state input matrix"
+            + f"has shape {state.shape}"
+        )
+    if op.shape[1] != state.shape[0]:
+        raise ValueError(f"incompatible shapes op:{op.shape} and state:{state.shape}")
+    if op.shape[0] != op.shape[1]:
+        raise ValueError(
+            "Expect_super only works for square op matrices. The"
+            f"provided op has shape: {op.shape}"
```

Comment:
there's spaces missing in the concatenation of these two (top and bottom) strings.  Also, you don't need the `+` in the top one - string literals, including f-strings, get concatenated within the AST generation.